cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1208,41 +1211,82 @@ CUDF_KERNEL void __launch_bounds__(1024)
+  static constexpr double trivial_case_cost_ratio = 0.1;
+  static constexpr double host_constant_overhead  = 5000;
+  static constexpr double cost_decay_exponent     = 4;
+
+  static CUDF_HOST_DEVICE double cost_factor(size_t input_size,
+                                             size_t output_size,
+                                             task_type task_type)
+  {
+    if (task_type == task_type::DECOMPRESSION) {
+      auto const compression_ratio = std::max(1., static_cast<double>(output_size) / input_size);
+      // When the compression ratio is one, the cost factor is the same as the copy cost ratio,
+      // meaning that the cost of decompressing the block is the same as the cost of copying it. The
+      // cost factor asymptotes to one as the compression ratio increases, meaning that the cost
+      // approaches the base cost of decompressing (which is a lot higher than the copy cost)
+      return 1. - (1. - trivial_case_cost_ratio) /
+                    cuda::std::pow(compression_ratio, cost_decay_exponent);
+    } else {
+      // We don't know the compression ratio for compression, so use a constant cost factor
+      return 1.;
+    }
   }
-}
 
-void gpu_copy_uncompressed_blocks(device_span<device_span<uint8_t const> const> inputs,
-                                  device_span<device_span<uint8_t> const> outputs,
-                                  rmm::cuda_stream_view stream)
-{
-  if (inputs.size() > 0) {
-    copy_uncompressed_kernel<<<inputs.size(), 1024, 0, stream.value()>>>(inputs, outputs);
+ public:
+  static CUDF_HOST_DEVICE double task_device_cost(size_t input_size,
+                                                  size_t output_size,
+                                                  task_type task_type)
+  {
+    return cost_factor(input_size, output_size, task_type) * input_size;
   }
-}
+
+  static CUDF_HOST_DEVICE double task_host_cost(size_t input_size,
+                                                size_t output_size,
+                                                double device_host_ratio,
+                                                task_type task_type)
+  {
+    CUDF_EXPECTS(device_host_ratio > 0, "device_host_ratio must be > 0");
+    // Cost to copy the block to host and back; NOTE: assumes that the copy throughput is the same
+    // as the decompression/compression throughput when the data is incompressible
+    auto const copy_cost = trivial_case_cost_ratio * (input_size + output_size);
+    return (cost_factor(input_size, output_size, task_type) * input_size + copy_cost) /
+             device_host_ratio +
+           host_constant_overhead;
+  }
+};
 
 sorted_codec_parameters sort_tasks(device_span<device_span<uint8_t const> const> inputs,
                                    device_span<device_span<uint8_t> const> outputs,
                                    rmm::cuda_stream_view stream,
+                                   task_type task_type,
```

Comment:
This should come before stream as stream and mr should be the last two args per our convention. Bonus and completely optional if we could make this a templated param